src/app/domains/domains.service.spec.ts

```diff
@@ -58,51 +89,86 @@ describe('DomainsService', () => {
 
     service.getRecord('testDomain').subscribe(response => {
       expect(response).toEqual(domainResponse)
-      done()
     })
   })
 
-  it('should return the updated domain details when a domain is updated', (done: DoneFn) => {
-    const domainReq = {
-      profileName: 'testDomain',
-      domainSuffix: 'testDomain.com',
-      provisioningCert: 'domainCertLongText',
-      provisioningCertPassword: 'password',
-      provisioningCertStorageFormat: 'string'
+  it('should return errors on a single domain object detail when a single record is requested', () => {
+    const error = {
+      status: 404,
+      message: 'Not Found'
     }
 
+    httpClientSpy.get.and.returnValue(throwError(error))
+
+    service.getRecord('testDomain').subscribe(() => {}, err => {
+      expect(error.status).toBe(err[0].status)
+    })
+  })
+
+  it('should return the updated domain details when a domain is updated', () => {
     httpClientSpy.patch.and.returnValue(of(domainReq))
 
     service.update(domainReq).subscribe(response => {
       expect(response).toEqual(domainReq)
-      done()
     })
   })
 
-  it('should return the created domain details when a domain is created', (done: DoneFn) => {
-    const domainReq = {
-      profileName: 'testDomain',
-      domainSuffix: 'testDomain.com',
-      provisioningCert: 'domainCertLongText',
-      provisioningCertPassword: 'password',
-      provisioningCertStorageFormat: 'string'
+  it('should return errors on updated domain details when a domain is updated', () => {
+    const error = {
+      status: 404,
+      message: 'Not Found'
     }
 
+    httpClientSpy.patch.and.returnValue(throwError(error))
+
+    service.update(domainReq).subscribe(() => {}, err => {
+      expect(error.status).toBe(err[0].status)
+    })
+  })
+
+  it('should return the created domain details when a domain is created', () => {
     httpClientSpy.post.and.returnValue(of(domainReq))
 
     service.create(domainReq).subscribe(response => {
       expect(response).toEqual(domainReq)
-      done()
     })
   })
 
-  it('should delete the domain when a delete request is sent', (done: DoneFn) => {
+  it('should throw error on created domain details when a domain is created', () => {
+    const error = {
+      status: 404,
+      message: 'Not Found'
+    }
+
+    httpClientSpy.post.and.returnValue(throwError(error))
+
+    service.create(domainReq).subscribe(() => {}, err => {
+      expect(error.status).toBe(err[0].status)
+    })
+  })
+
+  it('should delete the domain when a delete request is sent', () => {
     const domainName = 'testDomain'
 
     httpClientSpy.delete.and.returnValue(of({}))
 
     service.delete(domainName).subscribe(() => {
-      done()
+    })
+
+    expect(httpClientSpy.delete.calls.count()).toEqual(1)
+  })
+
+  it('should throw error when a delete request is sent', () => {
+    const domainName = 'testDomain'
+    const error = {
```

Comment:
same error is created 3 times, can be created once up above

Reply:
done